op-challenger/game/fault/types/game.go

```diff
@@ -112,62 +106,40 @@ func (g *gameState) Put(claim Claim) error {
 	if parent == nil {
 		return errors.New("no parent claim")
 	}
-	parent.children = append(parent.children, makeClaimEntry(claim))
-	claimWithExtension := &extendedClaim{
-		self:     claim,
-		children: make([]claimEntry, 0),
-	}
-	g.claims[makeClaimEntry(claim)] = claimWithExtension
-	g.contractIndicies[claim.ContractIndex] = claimWithExtension
+
+	g.claims = append(g.claims, claim)
```

Comment:
This would break claims ordering if Put wasn't called in the right order. This could easily be misused as the caller needs to make sure that they're Putting claims in order.

Perhaps we still need the old map for efficient updates to the game state.

Reply:
I was thinking the next thing we should do is delete `Put` and `PutAll`. Production code never actually uses them.

Reply:
Have pushed up an additional commit that removes `Put` and `PutAll`. Might cause a few merge conflicts with the deep position changes but should be ok.

Reply:
We aren't performing the same validation on inputs from the contracts we were in terms of checking the parent index lines up etc, but we really need to trust our L1 node or we're going to lose all our bonds anyway so I think I'm ok with that.  I could add them back into the NewGameState or the loader component if we think they're needed though.